config/config.go

```diff
@@ -52,8 +52,8 @@ func InitConfig() {
 	initProject()
 }
 
-func initCfg(path string, gettable bool, settable bool) cfg {
-	cfg := cfg{path, gettable, settable}
+func initCfg(path string, gettable bool, settable bool, setD bool, d string) cfg {
```

Comment:
@andscoop this method signature is starting to feel a little off. Same for the init block of callers above it.

What if we added a `NewCfg` method to `meta.go` that acts as a "constructor" and returns a new `cfg` object, with default values. Seems like a lot of this the invocations of this method use the same defaults. Maybe we could cut out some boilerplate, add some default values, and let callers of `NewCfg` directly override specific values by name. Something like

```
PostgresUser := NewCfg(cfg{Name: "postgres.user", Default: "postgres"})
```

Really comes down to moving the `initCfg` function to live with the `cfg` type definition and using the `NewXXX` pattern. Rather than having a seemingly arbitrary list of parameters to pass every time, we can set default values, that are overridden specifically by name, rather than just appearing in a list.

Reply:
Maybe we should rename `meta.go` to `types.go` as well. Then we could have multiple `Cfg` types that maybe share some group of defaults. For example, `GettableCfg`, `SetableCfg`. Not sure if multiple types is really needed yet.

Also, do we need a `setD` param here? Could we get away with just checking if the `Cfg` has a `Default` value set when created and call `setDefault` if it's not `nil` or an empty string?